doc/source/dev/contributor/recommended_development_setup.rst

```diff
@@ -0,0 +1,54 @@
+:orphan:
+
+.. _recommended-development-setup:
+
+=============================
+Recommended development setup
+=============================
+
+*This document does not include detailed explanations. For more step-by-step
+procedures, see* :ref:`quickstart-mac` *or* :ref:`building`.
+
+Since SciPy contains parts written in C, C++, and Fortran that need to be
+compiled before use, make sure you have the necessary compilers and Python
+development headers installed.  Having compiled code also means that importing
+SciPy from the development sources needs some additional steps, which are
+explained below.
+
+First fork a copy of the main SciPy repository in Github onto your own
+account and then create your local repository via::
```

Comment:
For now I did that to this document since it already has the `$` at the beginning of each line. Let's create a separate issue if it's important to add this to all of the documentation.